File: Builder/PizzaExample/BuilderPizza/PizzaBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.ConstrainedExecution;
using System.Text;
using System.Threading.Tasks;

namespace BuilderPizza
{
    public class PizzaBuilder : IPizzaBuilder
    {
        private Pizza pizza;

        public PizzaBuilder()
        {
            Reset();
        }
        public void Reset()
        {
            pizza = new Pizza();
        }

        public void SetDough(string dough) => pizza.Dough = dough;
        public void SetSauce(string sauce) => pizza.Sauce = sauce;
        public void SetCheese(string cheese) => pizza.Cheese = cheese;
        public void SetTopping(string topping) => pizza.Topping = topping;
        public void SetSize(int size) =>    pizza.Size = size;
        public Pizza GetProduct()
        {
            Pizza result = pizza;
            Reset();
            return result;
        }
    }
}

File: Builder/Java/Pizza/src/PizzaBuilder.java
public class PizzaBuilder implements IPizzaBuilder {
    private Pizza pizza;

    public PizzaBuilder() {
        reset();
    }

    @Override
    public void reset() {
        this.pizza = new Pizza();
    }

    @Override
    public void setDough(String dough) {
        pizza.setDough(dough);
    }

    @Override
    public void setSauce(String sauce) {
        pizza.setSauce(sauce);
    }

    @Override
    public void setCheese(String cheese) {
        pizza.setCheese(cheese);
    }

    @Override
    public void setTopping(String topping) {
        pizza.setTopping(topping);
    }

    @Override
    public void setSize(Integer size) {
        pizza.setSize(size);
    }

    @Override
    public Pizza getProduct() {
        Pizza result = this.pizza;
        reset();
        return result;
    }
}